test/functional/readpreference.test.js

```diff
@@ -78,6 +78,50 @@ describe('ReadPreference', function () {
     });
   });
 
+  describe('fromOptions factory method', () => {
```

Comment:
Some meta test organizing, we are gradually improving the placement of tests so I think it would make sense to take this block and the `::constructor` one out of here and move them to a new file: `test/unit/read_preference.test.ts`. They don't depend on being connected to a MongoDB server like the other tests.

Reply:
Done!